client/src/components/InputsModal/InputsModal.tsx

```diff
@@ -56,6 +56,7 @@ const InputsModal: FC<InputsModalProps> = ({
     } catch (e) {
       // if JSON.parse fails, then assume field is not OAuth and move on
     }
+    if (input.type === 'radio') return false; // radio inputs will always be required and have a default value
```

Comment:
Is this true?

I see an of a non-required radio button here: https://github.com/inferno-framework/smart-app-launch-test-kit/blob/4ba8db2ccd1e517b92da0693641459f13a37d6d8/lib/smart_app_launch/app_redirect_test.rb#L28

There's no reason that a radio button has to be required with a default value so I'm not sure we want to enforce that and it would be more robust to enforce this on the test input itself than just the UI.

Reply:
The reasoning here is that radio buttons are always going to have values filled by virtue of being a radio button, i.e. a user can't really deselect a radio button once it's been selected (if this is not the case, then we should just be using a checkbox). To make this more explicit, a radio input with no default value will automatically select the first given value, so there's really no situation where this can be optional.

@Jammjammjamm Thoughts?

Reply:
> a radio input with no default value will automatically select the first given value

I don't think that's necessarily true and there's examples on the MDN docs for radio buttons where none are checked by default with the [`checked` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#attr-checked)

![Screen Shot 2022-02-08 at 12 59 11 PM](https://user-images.githubusercontent.com/41651655/153047711-a7dd8d27-07ad-4857-983c-b6f544517660.png)

Checkboxes are more typically for situations where you want to select one or more option vs radio buttons which only allow one to be selected. From an HTML stand point either could be optional though.

Reply:
I found this [codesandbox showing a material ui radio button with deselect](https://codesandbox.io/s/uncheck-radio-uuiy1?fontsize=14)

https://user-images.githubusercontent.com/41651655/153048570-65c6bf4e-bfc2-4f09-a442-4bd361b35cdd.mov

Reply:
There was a slack discussion about this on 1/28. My opinion is that a list of `n` radio buttons where no button might be selected is really a list of `n+1` radio buttons, except that one of the options is invisible and once deselected can never be selected again. I think that is dumb.

> > a radio input with no default value will automatically select the first given value

> I don't think that's necessarily true

No, that isn't automatically the case, but that is how it has been implemented here.

Reply:
I'm not necessarily saying we need to support optional radio buttons, but I'm concerned that a test can mark a input as optional and then the UI can make it required (especially since the CLI interface wouldn't similarly enforce this).

Reply:
I think the ability to deselect a radio button would be very unintuitive. I'd never try because that isn't how radio buttons generally work. I think if you want some sort of none-of-the-above option, it should be an explicit option.

Reply:
> except that one of the options is invisible and once deselected can never be selected again. I think that is dumb.

Agree its dumb unless deselect is implemented (which requires some JS AFAIK)

Reply:
I hadn't read through the slack conversation, so sorry that this rehashes some of that.

I'm concerned that a test can mark a input as optional and then the UI can make it required and what potential implications are for the test writer and CLI interface.

I think the problem might be more with Inferno Core allowing things like:

```ruby
     input :pkce_code_challenge_method,
          optional: true,
          title: 'PKCE Code Challenge Method',
          type: 'radio',
```

([source](https://github.com/inferno-framework/smart-app-launch-test-kit/blob/4ba8db2ccd1e517b92da0693641459f13a37d6d8/lib/smart_app_launch/app_redirect_test.rb#L27-L30))

which could surprise a test writer when they find out this argument is not optional in the UI, but is optional in the CLI.

Reply:
In any case, implementing deselect or otherwise should be a different ticket/PR -- this one only updates the required input conditional to match what's currently already happening. As is, the UI will assume a radio button with the first value automatically selected is still missing a value.